internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+package nginx
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2022-08-01/nginxconfiguration"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2022-08-01/nginxdeployment"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+const defaultConfigurationName = "default"
+
+type ConfigFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ProtectedFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ConfigurationModel struct {
+	Name              string          `tfschema:"name"` // always default
+	NginxDeploymentId string          `tfschema:"nginx_deployment_id"`
+	ConfigFile        []ConfigFile    `tfschema:"config_file"`
+	ProtectedFile     []ProtectedFile `tfschema:"protected_file"`
+	PackageData       string          `tfschema:"package_data"`
+	RootFile          string          `tfschema:"root_file"`
+}
+
+// ToSDKModel used in both Create and Update
+func (c ConfigurationModel) ToSDKModel() nginxconfiguration.NginxConfiguration {
+	req := nginxconfiguration.NginxConfiguration{
+		Name: pointer.FromString(c.Name),
+		Properties: &nginxconfiguration.NginxConfigurationProperties{
+			RootFile: pointer.FromString(c.RootFile),
+		},
+	}
+
+	var files []nginxconfiguration.NginxConfigurationFile
+	for _, file := range c.ConfigFile {
+		files = append(files, nginxconfiguration.NginxConfigurationFile{
+			Content:     pointer.FromString(file.Content),
+			VirtualPath: pointer.FromString(file.VirtualPath),
+		})
+	}
+	req.Properties.Files = &files
+
+	if len(c.ProtectedFile) > 0 {
+		var protectedFiles []nginxconfiguration.NginxConfigurationFile
+		for _, file := range c.ProtectedFile {
+			protectedFiles = append(protectedFiles, nginxconfiguration.NginxConfigurationFile{
+				Content:     pointer.FromString(file.Content),
+				VirtualPath: pointer.FromString(file.VirtualPath),
+			})
+		}
+		req.Properties.ProtectedFiles = &protectedFiles
+	}
+
+	if c.PackageData != "" {
+		req.Properties.Package = &nginxconfiguration.NginxConfigurationPackage{
+			Data: pointer.FromString(c.PackageData),
+		}
+	}
+
+	return req
+}
+
+type ConfigurationResource struct{}
+
+var _ sdk.Resource = (*ConfigurationResource)(nil)
+
+func (m ConfigurationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
we can validate the Deployment ID here: https://github.com/hashicorp/go-azure-sdk/blob/b5f8ed6d50deb207f33fbdc1ebc18d31536f913c/resource-manager/nginx/2022-08-01/nginxdeployment/id_nginxdeployment.go#L82